argcomplete/bash_completion.d/_python-argcomplete

```diff
@@ -139,9 +139,6 @@ _python_argcomplete_global() {
         __python_argcomplete_expand_tilde_by_ref executable
     else
         if [[ "$service" != "-default-" ]]; then
-            # TODO: this may not be sufficient - see https://zsh.sourceforge.io/Doc/Release/Completion-System.html
-            # May need to call _complete with avoid-completer=_python-argcomplete or something like that
-            _default
```

Comment:
@kislyuk This was added in https://github.com/kislyuk/argcomplete/commit/afdf75bc6f43f1d1584468665ea1147c22b9767c, but I'm not sure what the intent was. This was causing default completions to be mixed in with the returned completions in my tests, and removing it doesn't seem to have caused any of the existing tests to fail. If you remember what problem this was addressing, I can take another look into it.

Reply:
@evanunderscore please do not delete this without a replacement, and especially please do not delete the comment. This was added in response to https://github.com/kislyuk/argcomplete/issues/454. Specifically, it was found that the default completer interferes with zsh completion in the redirect target context (words after `>` or `2>`).

I didn't manage to write a test for this, and I since realized that this was a stopgap solution that will need to be further refined. For example, instead of calling _default unconditionally as a fallback here, we may need to call it only when certain contexts like `-redirect-` are listed in `$service` (see the link in the comment for a full list of contexts - there are others in the list that will almost certainly need similar treatment).

Reply:
Thanks for the issue link, I did search initially but wasn't able to find it.

I've pushed a fix in 1f6e66148ae8a4225481b4f3a9ae054d462cede0. Rather than handle all of the special contexts, I've updated the `compdef` command to only override `-default-` which leaves the other special contexts undisturbed. The tests are all passing and I've poked around manually in the shell for a bit, and it seems to be working as expected.

Are you happy with this fix and the comment I've left?

Reply:
Unfortunately the macos tests are failing and I don't have a good way to troubleshoot them. I can just disable the global zsh tests on macos (since they weren't running at all before I added them), but it is strange that they were all passing before I made the `compdef` change.

Reply:
The issue is that some of the runners already have `python-argcomplete` installed as a dependency of `pipx`, which explains why the failures seemed completely inconsistent - the tests were actually testing whichever version happened to be installed and not what was in the source. For now, I've added a step to uninstall it, but a better solution would be to revisit the change I suggested in #255. Without it, there is no good way to avoid this mismatch.

Reply:
> Rather than handle all of the special contexts, I've updated the compdef command to only override -default- which leaves the other special contexts undisturbed.

Sounds good, thanks.